src/ir_def/construct.ml

```diff
@@ -89,7 +89,11 @@ let primE prim es =
     | ICCallerPrim -> T.caller
     | ICStableRead t -> t
     | ICStableWrite _ -> T.unit
-    | IdxPrim -> T.(as_immut (as_array (List.hd es).note.Note.typ))
+    | IdxPrim
+    | DerefArrayOffset -> T.(as_immut (as_array (List.hd es).note.Note.typ))
```

Comment:
I wonder if this code will actually fail (and used to fail for IdxPrim) if the array argument has type None...

Reply:
(or is a generic type bounded by an array type)

Reply:
I remember writing a transformation a few weeks ago that called `is_array` on a value of type `None` and it returned false. I'll look it up. I assume that `as_array` would throw in that context.

Reply:
Do you mean, that the right semantics for `as_array` should return `None` for a `None` input?

Reply:
`as_array_sub` does precisely that: 78b8517f5